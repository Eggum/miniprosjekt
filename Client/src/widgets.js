// @flow

import * as React from 'react';
import {Component} from "react-simplified";
import { HashRouter, Route, NavLink } from 'react-router-dom';
import {createHashHistory} from "history";
import {Article, articleService} from "./services.js";


const history = createHashHistory();

export class Menu extends Component {


    categories : Category[] = [];

    mounted(){
        console.log("Meny mounted blir kjÃ¸rt");
        articleService.getCategories()
            .then((categories)=>{
                this.categories = categories})
            .catch((error: Error) => Alert.danger(error.message));
    }

    render() {
        return (
            <nav id="navigationBar" className="navbar navbar-expand-lg navbar-dark bg-dark sticky-top">
                <a className="navbar-brand" href="/">Home</a>
                <button className="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarNavAltMarkup"
                        aria-controls="navbarNavAltMarkup" aria-expanded="false" aria-label="Toggle navigation">
                    <span className="navbar-toggler-icon"/>
                </button>
                <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
                    <div className="navbar-nav">
                        <NavLink className="nav-link" to="/article/new">New article</NavLink>
                        <NavLink className="nav-link" to="/courses">Nothing</NavLink>
                    </div>
                    <div className="nav-item dropdown navbar-nav">
                        <a className="nav-link dropdown-toggle" href="#" id="navbarDropdown" role="button"
                           data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
                            Categories
                        </a>
                        <div className="dropdown-menu" aria-labelledby="navbarDropdown">
                            {this.categories.map((c, index) => (
                                <NavLink key={index} className="dropdown-item" to={"/article/" + c.category}>{c.category}</NavLink>
                            ))}
                            <div className="dropdown-divider"></div>
                        </div>
                    </div>
                    <form className="form-inline ml-auto" >
                        <input className="form-control mr-sm-2" type="search" placeholder="Search"
                               aria-label="Search" id="searchInput"/>
                        <button className="btn btn-outline-info my-2 my-sm-0" onClick={this.search}>Search</button>
                    </form>
                    <div className="navbar-nav ">
                        <NavLink className="nav-link" to="/courses">Nothing</NavLink>
                    </div>
                </div>
            </nav>
        );
    }

    search(){
        let input = document.getElementById('searchInput').value;
        if(input !== ""){

            //    Search.input = input;
            history.push('/search/' + input);

        }
    }
}
/*

                    <div className="carousel-item active">
                        <p className ="RENAME-ME-PLZ">Hei</p>
                    </div>
                    <div className="carousel-item">
                        <p className ="RENAME-ME-PLZ">Hei nr 2</p>
                    </div>
                    <div className="carousel-item">
                        <p className ="RENAME-ME-PLZ">Hei nr 3</p>
                    </div>
 */
export class Carousel extends Component<{children?: React.Node }>{
    render(){
        return(
            <div id="carouselExampleControls" className="carousel slide prettyCarousel" data-ride="carousel">
                <div className="carousel-inner">
                    {this.props.children}
                </div>
                <a className="carousel-control-prev" href="#carouselExampleControls" role="button" data-slide="prev">
                    <span className="carousel-control-prev-icon" aria-hidden="true"></span>
                    <span className="sr-only">Previous</span>
                </a>
                <a className="carousel-control-next" href="#carouselExampleControls" role="button" data-slide="next">
                    <span className="carousel-control-next-icon" aria-hidden="true"></span>
                    <span className="sr-only">Next</span>
                </a>
            </div>
        );
    }
}

export class Card extends Component<{ title: React.Node, image: React.Node, id: React.Node, alt: React.Node}> {
    render() {
        return (
            <div className="card card-pretty">
                <img src={this.props.image} className="card-img-top" alt={this.props.alt}/>
                <div className="card-body">
                    <h5 className="card-title">{this.props.title}</h5>
                    <NavLink to={'/article/' + +this.props.id} className="stretched-link">Les mer</NavLink>
                </div>
            </div>
        );
    }
}

export class ConfirmBox extends Component<{ modalId : React.Node, modalHeader : React.Node, modalBody : React.Node, onClick : () => mixed, children?: React.Node }>{
    render(){
        return(
            <div className="modal fade" id={this.props.modalId} tabIndex="-1" role="dialog"
                 aria-labelledby="exampleModalLabel" aria-hidden="true">
                <div className="modal-dialog" role="document">
                    <div className="modal-content">
                        <div className="modal-header">
                            <h5 className="modal-title" id="exampleModalLabel">{this.props.modalHeader}</h5>
                            <button type="button" className="close" data-dismiss="modal" aria-label="Close">
                                <span aria-hidden="true">&times;</span>
                            </button>
                        </div>
                        <div className="modal-body">
                            {this.props.modalBody}
                        </div>
                        <div className="modal-footer">
                            <button type="button" className="btn btn-secondary" data-dismiss="modal">No</button>
                            <button type="button" className="btn btn-primary" data-dismiss="modal" onClick={this.props.onClick}>Yes</button>
                        </div>
                    </div>
                </div>
            </div>
        )
    }
}


/**
 * Renders alert messages using Bootstrap classes.
 */
export class Alert extends Component {
    alerts: { id: number, text: React.Node, type: string }[] = [];
    static nextId = 0;

    render() {
        return (
            <>
                {this.alerts.map((alert, i) => (
                    <div key={alert.id} className={'alert alert-' + alert.type} role="alert">
                        {alert.text}
                        <button
                            type="button"
                            className="close"
                            onClick={() => {
                                this.alerts.splice(i, 1);
                            }}
                        >
                            &times;
                        </button>
                    </div>
                ))}
            </>
        );
    }

    static success(text: React.Node) {
        // To avoid 'Cannot update during an existing state transition' errors, run after current event through setTimeout
        setTimeout(() => {
            for (let instance of Alert.instances()) instance.alerts.push({ id: Alert.nextId++, text: text, type: 'success' });
        });
    }

    static info(text: React.Node) {
        // To avoid 'Cannot update during an existing state transition' errors, run after current event through setTimeout
        setTimeout(() => {
            for (let instance of Alert.instances()) instance.alerts.push({ id: Alert.nextId++, text: text, type: 'info' });
        });
    }

    static warning(text: React.Node) {
        // To avoid 'Cannot update during an existing state transition' errors, run after current event through setTimeout
        setTimeout(() => {
            for (let instance of Alert.instances()) instance.alerts.push({ id: Alert.nextId++, text: text, type: 'warning' });
        });
    }

    static danger(text: React.Node) {
        // To avoid 'Cannot update during an existing state transition' errors, run after current event through setTimeout
        setTimeout(() => {
            for (let instance of Alert.instances()) instance.alerts.push({ id: Alert.nextId++, text: text, type: 'danger' });
        });
    }
}
