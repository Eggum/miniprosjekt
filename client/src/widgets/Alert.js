import {Component} from "react-simplified";
import * as React from "react";

/**
 * Renders alert messages using Bootstrap classes.
 */
export class Alert extends Component {
    alerts: { id: number, text: React.Node, type: string }[] = [];
    static nextId = 0;

    render() {
        return (
            <>
                {this.alerts.map((alert, i) => (
                    <div className="alert-pretty" key={alert.id}>
                        <div
                            key={alert.id}
                            className={'alert alert-' + alert.type}
                            role="alert"
                        >
                            {alert.text}
                            <button
                                type="button"
                                className="close"
                                onClick={() => {
                                    this.alerts.splice(i, 1);
                                }}
                            >
                                &times;
                            </button>
                        </div>
                    </div>
                ))}
            </>
        );
    }

    static success(text: React.Node) {
        // To avoid 'Cannot update during an existing state transition' errors, run after current event through setTimeout
        setTimeout(() => {
            for (let instance of Alert.instances())
                instance.alerts.push({
                    id: Alert.nextId++,
                    text: text,
                    type: 'success'
                });
        });
    }

    static info(text: React.Node) {
        // To avoid 'Cannot update during an existing state transition' errors, run after current event through setTimeout
        setTimeout(() => {
            for (let instance of Alert.instances())
                instance.alerts.push({
                    id: Alert.nextId++,
                    text: text,
                    type: 'info'
                });
        });
    }

    static warning(text: React.Node) {
        // To avoid 'Cannot update during an existing state transition' errors, run after current event through setTimeout
        setTimeout(() => {
            for (let instance of Alert.instances())
                instance.alerts.push({
                    id: Alert.nextId++,
                    text: text,
                    type: 'warning'
                });
        });
    }

    static danger(text: React.Node) {
        // To avoid 'Cannot update during an existing state transition' errors, run after current event through setTimeout
        setTimeout(() => {
            for (let instance of Alert.instances())
                instance.alerts.push({
                    id: Alert.nextId++,
                    text: text,
                    type: 'danger'
                });
        });
    }
}