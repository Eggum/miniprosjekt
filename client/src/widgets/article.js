// @flow

import {Component} from "react-simplified";
import * as React from "react";
import {NavLink} from "react-router-dom";

export class Article extends Component<
> {
    render() {
        return null;
    }
}