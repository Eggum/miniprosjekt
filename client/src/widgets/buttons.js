// @flow

import * as React from 'react';
import { Component } from 'react-simplified';

class ButtonPrimary extends Component<{
    onClick: () => mixed,
    children?: React.Node
}> {
    render() {
        return (
            <button
                type="button"
                className="btn btn-primary"
                onClick={this.props.onClick}
            >
                {this.props.children}
            </button>
        );
    }
}

class ButtonDanger extends Component<{
    onClick: () => mixed,
    children?: React.Node
}> {
    render() {
        return (
            <button
                type="button"
                className="btn btn-danger"
                onClick={this.props.onClick}
            >
                {this.props.children}
            </button>
        );
    }
}

class ButtonDangerSmallFloatRight extends Component<{
    onClick: () => mixed,
    children?: React.Node
}> {
    render() {
        return (
            <button
                type="button"
                className="btn btn-danger btn-sm float-right"
                onClick={this.props.onClick}
            >
                {this.props.children}
            </button>
        );
    }
}

class ButtonRound extends Component<{
    onClick: () => mixed,
    children?: React.Node
}> {
    render() {
        return (
            <button
                type="button"
                className="buttonRound"
                onClick={this.props.onClick}
            >
                {this.props.children}
            </button>
        );
    }
}

class ButtonModalDanger extends Component<{
    dataTarget: string,
    children?: React.Node
}> {
    render() {
        return (
            <button
                type="button"
                className="btn btn-danger"
                data-toggle="modal"
                data-target={'#' + this.props.dataTarget}
            >
                {this.props.children}
            </button>
        );
    }
}

class ButtonSubmit extends Component<{ children?: React.Node }> {
    render() {
        return (
            <button className="btn btn-primary" type="submit">
                {this.props.children}
            </button>
        );
    }
}

export class Button {
    static Round = ButtonRound;
    static Danger = ButtonDanger;
    static Primary = ButtonPrimary;
    static ModalDanger = ButtonModalDanger;
    static DangerSmallRight = ButtonDangerSmallFloatRight;
    static Submit = ButtonSubmit;
}
