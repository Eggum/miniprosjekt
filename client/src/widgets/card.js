// @flow

import {Component} from "react-simplified";
import * as React from "react";
import {NavLink} from "react-router-dom";

export class Card extends Component<{
    title: React.Node,
    image: React.Node,
    id: React.Node,
    alt: React.Node
}> {
    render() {
        return (
            <div className="card card-pretty">
                <img
                    src={this.props.image}
                    className="card-img-top"
                    alt={this.props.alt}
                />
                <div className="card-body">
                    <h5 className="card-title">{this.props.title}</h5>
                    <NavLink
                        to={'/article/' + +this.props.id}
                        className="stretched-link"
                    >
                        Read more
                    </NavLink>
                </div>
            </div>
        );
    }
}