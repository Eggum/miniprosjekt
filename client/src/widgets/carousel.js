// @flow

import {Component} from "react-simplified";
import * as React from "react";

export class Carousel extends Component<{ children?: React.Node }> {
    render() {
        return (
            <div
                id="carouselExampleControls"
                className="carousel slide prettyCarousel"
                data-ride="carousel"
            >
                <div className="carousel-inner">{this.props.children}</div>
                <a
                    className="carousel-control-prev"
                    href="#carouselExampleControls"
                    role="button"
                    data-slide="prev"
                >
                    <span
                        className="carousel-control-prev-icon"
                        aria-hidden="true"
                    />
                    <span className="sr-only">Previous</span>
                </a>
                <a
                    className="carousel-control-next"
                    href="#carouselExampleControls"
                    role="button"
                    data-slide="next"
                >
                    <span
                        className="carousel-control-next-icon"
                        aria-hidden="true"
                    />
                    <span className="sr-only">Next</span>
                </a>
            </div>
        );
    }
}