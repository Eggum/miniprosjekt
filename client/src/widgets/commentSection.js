// @flow

import * as React from 'react';
import { Component } from 'react-simplified';
import { Comment } from '../services.js';
import { Button } from './buttons';
import { connect } from 'react-redux';

/**
 * Displays the comment section. Used in view article.
 * If the user is logged in, the user can delete own comments.
 * Uses bootstrap for styling.
 */

// The states that are going to be mapped to the component as properties.
function mapStateToProps(state) {
    return {
        stateName: state.name,
        isLogged: state.isLogged,
        stateID: state.id
    };
}

class CommentSectionComp extends Component<{
    comments: Array<Comment>,
    newComment: Comment,
    onClick: () => mixed,
    onDelete: (c: Comment) => mixed,
    isLogged: boolean,
    stateName: string,
    stateID: number
}> {
    render() {
        return (
            <div className="commentSection">
                <h2>Comments:</h2>
                {this.props.comments.map((c: Comment) => (
                    <div className="card bg-light mb-3" key={c.id}>
                        <div className="card-header">
                            <h5 className="card-title">{c.username}</h5>
                        </div>
                        <div className="card-body">
                            <p className="card-text">{c.text}</p>
                            <small className="card-subtitle mb-2 text-muted">
                                {new Date(c.creation_date).toLocaleString()}
                            </small>
                            {this.props.isLogged &&
                            this.props.stateID === c.creator ? (
                                <Button.DangerSmallRight
                                    onClick={() => this.props.onDelete(c)}
                                >
                                    Delete comment
                                </Button.DangerSmallRight>
                            ) : null}
                        </div>
                    </div>
                ))}
                <label htmlFor="newComment">New comment</label>
                {this.props.isLogged ? (
                    <h5>{this.props.stateName}</h5>
                ) : (
                    <h5>Anonym</h5>
                )}
                <input
                    required
                    type="text"
                    className="form-control"
                    id="newComment"
                    placeholder="..."
                    onChange={(
                        event: SyntheticInputEvent<HTMLInputElement>
                    ) => {
                        if (this.props.newComment)
                            this.props.newComment.text = event.target.value;
                    }}
                />
                <Button.Primary onClick={this.props.onClick}>
                    Publish comment
                </Button.Primary>
            </div>
        );
    }
}

/*
Connects the state to the component.
The original component is not changed. Instead, connect function returns a new, connected component class that wraps the component passed in.
 */
export const CommentSection = connect(mapStateToProps)(CommentSectionComp);
