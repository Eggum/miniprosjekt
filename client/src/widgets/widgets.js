// @flow

import * as React from 'react';
import {Component} from "react-simplified";
import { HashRouter, Route, NavLink } from 'react-router-dom';
import {createHashHistory} from "history";
import {Category, Article, articleService} from "../services.js";
import {Button} from "./buttons";


const history = createHashHistory();

/*

                    <div className="carousel-item active">
                        <p className ="RENAME-ME-PLZ">Hei</p>
                    </div>
                    <div className="carousel-item">
                        <p className ="RENAME-ME-PLZ">Hei nr 2</p>
                    </div>
                    <div className="carousel-item">
                        <p className ="RENAME-ME-PLZ">Hei nr 3</p>
                    </div>
 */
export class Carousel extends Component<{children?: React.Node }>{
    render(){
        return(
            <div id="carouselExampleControls" className="carousel slide prettyCarousel" data-ride="carousel">
                <div className="carousel-inner">
                    {this.props.children}
                </div>
                <a className="carousel-control-prev" href="#carouselExampleControls" role="button" data-slide="prev">
                    <span className="carousel-control-prev-icon" aria-hidden="true"/>
                    <span className="sr-only">Previous</span>
                </a>
                <a className="carousel-control-next" href="#carouselExampleControls" role="button" data-slide="next">
                    <span className="carousel-control-next-icon" aria-hidden="true"/>
                    <span className="sr-only">Next</span>
                </a>
            </div>
        );
    }
}

export class Form extends Component<{ article: Article, dataTarget : string,  onSubmit : (event: SyntheticInputEvent<HTMLFormElement>) => mixed}>{
    categories : Category[] = [];

    mounted(){
        articleService.getCategories()
            .then((categories)=>{
                this.categories = categories})
            .catch((error: Error) => Alert.danger(error.message));
    }

    render(){
        return(
            <form onSubmit={this.props.onSubmit} className="needs-validation">
                <div className="form-group">
                    <label htmlFor="title">Title</label>
                    <input required type="text" className="form-control" id="title" maxLength="30" value={this.props.article.title} placeholder="Write your title here"
                           onChange={(event: SyntheticInputEvent<HTMLInputElement>) => {
                               if (this.props.article) this.props.article.title = event.target.value;
                           }}/>
                    <div className="invalid-feedback">
                        Please provide a valid zip.
                    </div>
                </div>
                <div className="form-group">
                    <label htmlFor="image">Article image</label>
                    <input required type="text" className="form-control" id="image" value={this.props.article.image} placeholder="Write image url"
                           onChange={(event: SyntheticInputEvent<HTMLInputElement>) => {
                               if (this.props.article) this.props.article.image = event.target.value;
                           }}/>
                </div>
                <div className="form-group">
                    <label htmlFor="imageText">Image text</label>
                    <input required type="text" className="form-control" maxLength="50" id="imageText" value={this.props.article.image_text} placeholder="Write image text"
                           onChange={(event: SyntheticInputEvent<HTMLInputElement>) => {
                               if (this.props.article) this.props.article.image_text = event.target.value;
                           }}/>
                </div>
                <div className="form-group">
                    <label htmlFor="alt">Image alt</label>
                    <input required type="text" className="form-control" id="alt" maxLength="50" value={this.props.article.alt}  placeholder="image alt"
                           onChange={(event: SyntheticInputEvent<HTMLInputElement>) => {
                               if (this.props.article) this.props.article.alt = event.target.value;
                           }}/>
                </div>
                <div className="form-group form-check">
                    <input type="checkbox" className="form-check-input" id="importance" onChange={(event: SyntheticInputEvent<HTMLInputElement>) => {
                        if (this.props.article) if (event.target.checked){
                            this.props.article.importance = 1;
                        } else {
                            this.props.article.importance = 2;
                        }
                    }}/>
                        <label className="form-check-label" htmlFor="importance">Check this if article is IMPORTANT</label>
                </div>

                <label className="radio-inline">Category</label>

                <div className="from-group">
                    {this.categories.map((c, index) => (
                        this.props.article.category === c.category ?
                            <div className="form-check form-check-inline" key={index}>
                                <input checked className="form-check-input" type="radio" name="inlineRadioOptions" id={"radioButtonCategory" + index}
                                       value={c.category} onChange={(event: SyntheticInputEvent<HTMLInputElement>) => {
                                    if (this.props.article) this.props.article.category = event.target.value;}}/>
                                <label className="form-check-label" htmlFor={"radioButtonCategory" + index}>{c.category}</label>
                            </div>
                            :
                            <div className="form-check form-check-inline" key={index}>
                                <input className="form-check-input" type="radio" name="inlineRadioOptions" id={"radioButtonCategory" + index}
                                       value={c.category} onChange={(event: SyntheticInputEvent<HTMLInputElement>) => {
                                    if (this.props.article) this.props.article.category = event.target.value;}}/>
                                <label className="form-check-label" htmlFor={"radioButtonCategory" + index}>{c.category}</label>
                            </div>
                    ))}
                </div>
                <div className="form-group">
                </div>
                <div className="form-group">
                    <label htmlFor="articleText">Text</label>
                    <textarea required className="form-control" rows="20" value={this.props.article.text} placeholder="Article text here" id="articleText"
                              onChange={(event: SyntheticInputEvent<HTMLInputElement>) => {
                                  if (this.props.article) this.props.article.text = event.target.value;
                              }}/>
                </div>
                <Button.ModalDanger  dataTarget={this.props.dataTarget}>Cancel</Button.ModalDanger>
                <button className="btn btn-primary" type="submit">Save</button>
            </form>
        )
    }
}

export class Card extends Component<{ title: React.Node, image: React.Node, id: React.Node, alt: React.Node}> {
    render() {
        return (
            <div className="card card-pretty">
                <img src={this.props.image} className="card-img-top" alt={this.props.alt}/>
                <div className="card-body">
                    <h5 className="card-title">{this.props.title}</h5>
                    <NavLink to={'/article/' + +this.props.id} className="stretched-link">Les mer</NavLink>
                </div>
            </div>
        );
    }
}

export class ConfirmBox extends Component<{ modalId : React.Node, modalHeader : React.Node, modalBody : React.Node, onClick : () => mixed, children?: React.Node }>{
    render(){
        return(
            <div className="modal fade" id={this.props.modalId} tabIndex="-1" role="dialog"
                 aria-labelledby="exampleModalLabel" aria-hidden="true">
                <div className="modal-dialog" role="document">
                    <div className="modal-content">
                        <div className="modal-header">
                            <h5 className="modal-title" id="exampleModalLabel">{this.props.modalHeader}</h5>
                            <button type="button" className="close" data-dismiss="modal" aria-label="Close">
                                <span aria-hidden="true">&times;</span>
                            </button>
                        </div>
                        <div className="modal-body">
                            {this.props.modalBody}
                        </div>
                        <div className="modal-footer">
                            <button type="button" className="btn btn-secondary" data-dismiss="modal">No</button>
                            <button type="button" className="btn btn-primary" data-dismiss="modal" onClick={this.props.onClick}>Yes</button>
                        </div>
                    </div>
                </div>
            </div>
        )
    }
}


/**
 * Renders alert messages using Bootstrap classes.
 */
export class Alert extends Component {
    alerts: { id: number, text: React.Node, type: string }[] = [];
    static nextId = 0;

    render() {
        return (
            <>
                {this.alerts.map((alert, i) => (
                    <div className="alert-pretty" key={alert.id}>
                        <div key={alert.id} className={'alert alert-' + alert.type} role="alert">
                            {alert.text}
                            <button
                                type="button"
                                className="close"
                                onClick={() => {
                                    this.alerts.splice(i, 1);
                                }}
                            >
                                &times;
                            </button>
                        </div>
                    </div>
                ))}
            </>
        );
    }

    static success(text: React.Node) {
        // To avoid 'Cannot update during an existing state transition' errors, run after current event through setTimeout
        setTimeout(() => {
            for (let instance of Alert.instances()) instance.alerts.push({ id: Alert.nextId++, text: text, type: 'success' });
        });
    }

    static info(text: React.Node) {
        // To avoid 'Cannot update during an existing state transition' errors, run after current event through setTimeout
        setTimeout(() => {
            for (let instance of Alert.instances()) instance.alerts.push({ id: Alert.nextId++, text: text, type: 'info' });
        });
    }

    static warning(text: React.Node) {
        // To avoid 'Cannot update during an existing state transition' errors, run after current event through setTimeout
        setTimeout(() => {
            for (let instance of Alert.instances()) instance.alerts.push({ id: Alert.nextId++, text: text, type: 'warning' });
        });
    }

    static danger(text: React.Node) {
        // To avoid 'Cannot update during an existing state transition' errors, run after current event through setTimeout
        setTimeout(() => {
            for (let instance of Alert.instances()) instance.alerts.push({ id: Alert.nextId++, text: text, type: 'danger' });
        });
    }
}
