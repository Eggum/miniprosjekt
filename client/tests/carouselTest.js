// @flow

import * as React from 'react';
import { shallow, mount, ShallowWrapper } from 'enzyme';
import { Carousel } from '../src/widgets/carousel';

/**
 * Carousel tests.
 */

describe('Carousel tests', () => {
    const wrapper: ShallowWrapper = shallow(<Carousel />);
    it('renders correctly', () => {
        expect(wrapper.debug()).toMatchSnapshot();
    });
});
