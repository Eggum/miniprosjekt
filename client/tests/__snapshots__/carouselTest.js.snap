// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel tests renders correctly 1`] = `
"<div id=\\"carouselExampleControls\\" className=\\"carousel slide prettyCarousel\\" data-ride=\\"carousel\\">
  <div className=\\"carousel-inner\\" />
  <a className=\\"carousel-control-prev\\" href=\\"#carouselExampleControls\\" role=\\"button\\" data-slide=\\"prev\\">
    <span className=\\"carousel-control-prev-icon\\" aria-hidden=\\"true\\" />
    <span className=\\"sr-only\\">
      Previous
    </span>
  </a>
  <a className=\\"carousel-control-next\\" href=\\"#carouselExampleControls\\" role=\\"button\\" data-slide=\\"next\\">
    <span className=\\"carousel-control-next-icon\\" aria-hidden=\\"true\\" />
    <span className=\\"sr-only\\">
      Next
    </span>
  </a>
</div>"
`;
